crypto/fipsmodule/bn/bytes.c

```diff
@@ -220,27 +235,36 @@ size_t BN_bn2bin(const BIGNUM *in, uint8_t *out) {
   return n;
 }
 
-int BN_bn2le_padded(uint8_t *out, size_t len, const BIGNUM *in) {
-  if (!fits_in_bytes(in->d, in->width, len)) {
-    return 0;
-  }
-
-  size_t num_bytes = in->width * BN_BYTES;
-  if (len < num_bytes) {
-    num_bytes = len;
+void bn_words_to_little_endian(uint8_t *out, size_t out_len, const BN_ULONG *in, const size_t in_len) {
+  // The caller should have selected an output length without truncation.
+  assert(fits_in_bytes(in, in_len, out_len));
+  size_t num_bytes = in_len * sizeof(BN_ULONG);
+  if (out_len < num_bytes) {
```

Comment:
should this be an error?

Reply:
This check is only performed in debug builds.  We could make the check in release builds, but it might add latency.

Reply:
This function is not exposed to consumers.  If the preconditions are not met, then it's a bug in our code that we're not performing the necessary checks.

Reply:
Sorry, I was looking at the `assert`. It's fine for `out_len < num_bytes`, it's not an error. 

`num_bytes` is initialized with the number of total bytes *available* from the input, but not every byte is significant. Not every byte from input needs to be copied -- we only need to copy `out_len` bytes to the output. When `out_len < num_bytes`, we set `num_bytes = out_len` because the logic below this requires `out_len >= num_bytes`.